tracing-subscriber/src/filter/directive.rs

```diff
@@ -38,6 +38,10 @@ pub(in crate::filter) trait Match {
     fn level(&self) -> &LevelFilter;
 }
 
+pub(in crate::filter) trait TargetMatch: Match {
+    fn cares_about_target(&self, to_check: &str) -> bool;
+}
```

Comment:
i don't really think i understand why this trait is necessary? since `directives_for_target` is only used with `StaticDirective` currently, why not just put that in the `DirectiveSet<StaticDirective>` impl?

Reply:
ah, great point